.github/workflows/main.yml

```diff
@@ -40,10 +42,22 @@ jobs:
            packaging
           python -m pip freeze
 
-      - name: Patch Package Versions
+      - name: Verify Changed files
+        uses: tj-actions/changed-files@v32
+        id: verify-changed-version
+        with:
+          files: |
+              odc/**
+              pyproject.toml
+              setup.py
+              setup.cfg
+
+      - name: Patch Package Versions when code change.
+        if: steps.verify-changed-version.outputs.any_changed == 'true'
         run: |
-          find . -name _version.py | xargs python ./scripts/patch_version.py \
-           $(git rev-parse --short HEAD)
+          # -1: auto increment of patch number
+          # other positive integers: set the patch number accordingly
+          python ./scripts/patch_version.py -1 ./odc/stats/_version.py
 
       - name: Build Clean Packages
         run: |
```

Comment:
Is there a missing step here that's supposed to push the wheels to a cache?

In the `test-wheels` job they're being pulled, but I can't see where from.

**Also**, there's a v3 of `actions/cache` that fixes a few deprecation warnings (and probably other things too).

Reply:
my understanding of `actions/cache`  -- it'll pull the cached file to `path` if it can find one and push from the `path` if it can't find one with `key`. So the lines here basically handle everything and it doesn't require explicit `push` or `pull` the cache.
https://github.com/opendatacube/odc-stats/blob/c62fabc497126331c4212e9c74fdf0517efe40fe/.github/workflows/main.yml#L26-L30

Reply:
As for the continuous release on every merge, based on the past behaviour how we update the version and build the docker image, most of time the version update and release happened when we wanted to change something in production, which basically implied every merge. We could potentially do it only when we think we need it, triggered by commit message syntax.